Add GCS Version 2 integration using Google cloud API (#12842)
Alluxio/alluxio: docs/en/ufs/GCS.md

For more information on GCS, please read its
[documentation](https://cloud.google.com/storage/docs/overview).

## Setup 

> Alluxio has two GCS integrations. The default GCS UFS module (GCS version 1) is implemented based on 

Reviewer: ```suggestion
Alluxio provides two ways to access GCS. The default GCS UFS module (GCS version 1) is implemented based on 
```

[jets3t](http://www.jets3t.org/) library which is design for AWS S3. 
Thus, it only accepts Google cloud storage interoperability access/secret keypair 
which allows full access to all Google cloud storages inside a Google cloud project.
No permission or access control can be placed on the interoperability keys.

> The conjuction of Google interoperability API and jets3t library also impact the performance of default GCS UFS module. 

Reviewer: ```suggestion
The conjunction of Google interoperability API and jets3t library also impact the performance of the default GCS UFS module. 
```

alluxio.master.mount.table.root.ufs=gs://GCS_BUCKET/GCS_DIRECTORY
```

> Choose your desired GCS implementation way and provide the corresponding Google credentials.

Reviewer: ```suggestion
Choose your preferred GCS UFS version and provide the corresponding Google credentials.
```
